pkg/internal/retry/retry.go

```diff
@@ -35,6 +36,9 @@ type temporary interface {
 
 // IsTemporary returns true if err implements Temporary() and it returns true.
 func IsTemporary(err error) bool {
+	if err == context.Canceled || err == context.DeadlineExceeded {
```

Comment:
You can drop the `context.Canceled` check, since tit will fail the typecheck below.